jv-oop-advanced: implement figures with OOP principles

tsuki99 opened the pull request.

tsuki99 committed:
solved task with interface implementation and abstraction usage

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **The test task has already been approved.**
> <sub>*[CODE: AAP]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **The test task has already been approved.**
> <sub>*[CODE: AAP]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **The test task has already been approved.**
> <sub>*[CODE: AAP]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **The test task has already been approved.**
> <sub>*[CODE: AAP]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

Elena-Bruyako commented on src/main/java/core/basesyntax/Circle.java:
```diff
@@ -0,0 +1,32 @@
+        return radius;
+    }
+
+    public void setRadius(int radius) {
```
remove unused method

Elena-Bruyako commented on src/main/java/core/basesyntax/Circle.java:
```diff
@@ -0,0 +1,32 @@
+        return Math.round(area * NUMBER_FOR_ROUND) / NUMBER_FOR_ROUND;
+    }
+
+    public double getRadius() {
```
replace get method after constructor

Elena-Bruyako commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -0,0 +1,11 @@
+
+public class ColorSupplier {
+    public String getRandomColor() {
+        int index = new Random().nextInt(Color.values().length);
```
make Random class-level variable

Elena-Bruyako commented on src/main/java/core/basesyntax/Figure.java:
```diff
@@ -0,0 +1,21 @@
+
+    public void setColor(String color) {
+        this.color = color;
+    }
```
```suggestion
    public String getColor() {
        return color;
    }
```

Elena-Bruyako commented on src/main/java/core/basesyntax/FigureSupplier.java:
```diff
@@ -0,0 +1,55 @@
+
+    private IsoscelesTrapezoid getRandomIsoscelesTrapezoid() {
+        return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_VALUE),
```
you can also create separate method to get random size `random.nextInt(MAX_VALUE)` to avoid duplication of code for each figure

Elena-Bruyako commented on src/main/java/core/basesyntax/Main.java:
```diff
@@ -0,0 +1,21 @@
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+                figures[i].draw();
+            }
```
```suggestion
                figures[i] = figureSupplier.getRandomFigure();
            } else {
                figures[i] = figureSupplier.getDefaultFigure();
            }
            figures[i].draw();
```

Elena-Bruyako reviewed (changes requested):
Good job! Let's improve your solution

tsuki99 committed:
removed unused methods
replaced get methods after constructors
made Random class lvl variable
created method for getting random size
moved call .draw() method a bit lower
